tests/mainnet/metaregistry/integrative/test_getters.py

```diff
@@ -0,0 +1,793 @@
+        else:
+            curve_contract = curve_pool_v2(pool)
+            actual_output = [
+                curve_contract.fee(),
+                curve_contract.admin_fee(),
+                curve_contract.mid_fee(),
+                curve_contract.out_fee(),
+            ]
+    except brownie.exceptions.VirtualMachineError:
+        actuals_reverts = True
+
+    if actuals_reverts:
+        with brownie.reverts():
+            metaregistry.get_fees(pool)
+    else:
+        metaregistry_output = metaregistry.get_fees(pool)
+        for j, output in enumerate(actual_output):
+            assert output == metaregistry_output[j]
+
+
+@pytest.mark.parametrize("pool_id", range(MAX_POOLS))
+def test_get_pool_name(metaregistry, registry_pool_index_iterator, pool_id):
+
+    skip_if_pool_id_gt_max_pools_in_registry(pool_id, registry_pool_index_iterator)
+
+    registry_id, registry_handler, registry, pool = registry_pool_index_iterator[pool_id]
+
+    # we skip if pool was registered: this is because some pools are
+    # registered in the stable registry but are from the stable factory.
+    # skipping does not skip the test: since metaregistry chooses stable
+    # factory as the registry handler and not the registry in these cases
+    if check_pool_already_registered(metaregistry, pool, registry_handler):
+        pytest.skip()
+
+    if registry_id in [
+        METAREGISTRY_STABLE_REGISTRY_HANDLER_INDEX,
+        METAREGISTRY_CRYPTO_REGISTRY_HANDLER_INDEX,
+    ]:
+
+        actual_output = registry.get_pool_name(pool)
+
+    elif registry_id == METAREGISTRY_CRYPTO_FACTORY_HANDLER_INDEX:
+
+        actual_output = brownie.interface.ERC20(registry.get_token(pool)).name()
+
+    else:
+
+        actual_output = brownie.interface.ERC20(pool).name()
+
+    metaregistry_output = metaregistry.get_pool_name(pool)
+    assert actual_output == metaregistry_output
+
+
+def _get_gauges_actual(registry, registry_id, pool):
+
+    if registry_id in [
+        METAREGISTRY_STABLE_REGISTRY_HANDLER_INDEX,
+        METAREGISTRY_CRYPTO_REGISTRY_HANDLER_INDEX,
+    ]:
+
+        actual_output = registry.get_gauges(pool)
+
+    else:
+
+        gauge = registry.get_gauge(pool)
+        actual_output = (
+            [gauge] + [brownie.ZERO_ADDRESS] * 9,
+            [gauge_controller().gauge_types(gauge)] + [0] * 9,
+        )
+
+    return actual_output
+
+
+@pytest.mark.parametrize("pool_id", range(MAX_POOLS))
+def test_get_gauges(metaregistry, registry_pool_index_iterator, pool_id):
+
+    skip_if_pool_id_gt_max_pools_in_registry(pool_id, registry_pool_index_iterator)
+
+    registry_id, registry_handler, registry, pool = registry_pool_index_iterator[pool_id]
+
+    # we skip if pool was registered: this is because some pools are
+    # registered in the stable registry but are from the stable factory.
+    # skipping does not skip the test: since metaregistry chooses stable
+    # factory as the registry handler and not the registry in these cases
```

Comment:
just add that comment in the `check_pool_already_registered` function instead of repeating it in every test

Reply:
done